HIRS_AttestationCA/src/main/java/hirs/attestationca/AbstractAttestationCertificateAuthority.java

```diff
@@ -492,6 +514,15 @@ private IdentityResponseEnvelope generateIdentityResponseEnvelopeAndStoreIssuedC
             Set<PlatformCredential> platformCredentials = parsePcsFromIdentityClaim(claim,
                     endorsementCredential);
 
+            // Parse through the Provisioner supplied TPM Quote and pcr values
+            // these fields are optional
+            if (request.getQuote() != null && !request.getQuote().isEmpty()) {
```

Comment:
getQuote() returns a string, it will throw and exception if you try to reference a method from a null value.  This was the last change I made so that if the optional get quote field hasn't been sent, it doesn't crash.

Reply:
I had believed that String.isEmpty() also checked for null values but this is not the case.  Code looks good here.